### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by dwbutler** [codecov/codecov-ruby#14]
Today I started seeing errors from Codecov in my Travis CI build. (See attached.) There is no way to rescue the error. The codecov gem should rescue errors and output a warning, instead of causing the build to fail.

```
<!DOCTYPE html>
    <html>
    <head>
      <meta name="viewport" content="width=device-width, initial-scale=1">
      <style type="text/css">
        html, body, iframe { margin: 0; padding: 0; height: 100%; }
        iframe { display: block; width: 100%; border: none; }
      </style>
    <title>Application Error</title>
    </head>
    <body>
      <iframe src="https://d234q63orb21db.cloudfront.net/cd118a714294f80d079e8514f43792f519841d20/media/html/error.html">
        <p>Application Error</p>
      </iframe>
    </body>
    </html>
/home/travis/.rvm/gems/ruby-2.0.0-p598/gems/json-2.0.1/lib/json/common.rb:156:in `parse': 740: unexpected token at '<!DOCTYPE html> (JSON::ParserError)
    <html>
    <head>
      <meta name="viewport" content="width=device-width, initial-scale=1">
      <style type="text/css">
        html, body, iframe { margin: 0; padding: 0; height: 100%; }
        iframe { display: block; width: 100%; border: none; }
      </style>
    <title>Application Error</title>
    </head>
    <body>
      <iframe src="https://d234q63orb21db.cloudfront.net/cd118a714294f80d079e8514f43792f519841d20/media/html/error.html">
        <p>Application Error</p>
      </iframe>
    </body>
    </html>'
    from /home/travis/.rvm/gems/ruby-2.0.0-p598/gems/json-2.0.1/lib/json/common.rb:156:in `parse'
    from /home/travis/.rvm/gems/ruby-2.0.0-p598/gems/codecov-0.1.5/lib/codecov.rb:220:in `format'
    from /home/travis/.rvm/gems/ruby-2.0.0-p598/gems/simplecov-0.12.0/lib/simplecov/result.rb:47:in `format!'
    from /home/travis/.rvm/gems/ruby-2.0.0-p598/gems/simplecov-0.12.0/lib/simplecov/configuration.rb:172:in `block in at_exit'
    from /home/travis/.rvm/gems/ruby-2.0.0-p598/gems/simplecov-0.12.0/lib/simplecov/defaults.rb:67:in `call'
    from /home/travis/.rvm/gems/ruby-2.0.0-p598/gems/simplecov-0.12.0/lib/simplecov/defaults.rb:67:in `block in <top (required)>'
```

**Comment by stevepeak** [comment]
Hey @dwbutler there was a Travis API outage this morning which may have caused this. Uploads were essentially rejected because the API was returning 500's.

Have you seen this issue in recent builds?

**Comment by dwbutler** [comment]
Yes, there were outages across the board due to outages in Github. It hasn't happened since this morning. It never happened before and maybe it will never happen again.

The issue is that a failure to report results to CodeCov should not cause a CI build to fail. It's not a problem with the code under test. And there isn't any way to work around the issue (as far as I can tell).

**Comment by stevepeak** [comment]
Was this the ruby uploader? https://github.com/codecov/codecov-ruby

**Comment by dwbutler** [comment]
Yes, the backtrace I attached above implicates the `codecov` gem.

**Comment by schasse** [comment]
happens again on our builds today.

**Comment by flockom** [comment]
Is there a recommended way to rescue this? This is also breaking our builds on circle. Ideally we aren't monkey patching your gem

**Comment by stevepeak** [comment]
Sorry about this issue. Having a hell-of-a-day today. My sincerest apologies for causing your builds to fail.

I'll see about patching the gem now.

**Comment by stevepeak** [comment]
Does this commit work? (Sorry not a Ruby programmer)

**Comment by flockom** [comment]
@stevepeak I tested this but it looks like the reports are submitting successfully now. Looks good to me though 👍

**Comment by dwbutler** [comment]
@stevepeak `rescue => err` is the recommended way to catch application errors. Otherwise looks good.

**Comment by ianjepperson** [comment]
@stevepeak Thank you, we had this problem today too.  Codecov is quite useful but not critical for our deployment.

**Comment by schasse** [comment]
Thanks for fixing this!

**Comment by stevepeak** [comment]
Fixed and deployed in `0.1.10`. Thank you
